### ISSUE TITLE
Details of the training of the rationale generator

### ISSUE THREAD

**Issue by Yzmshjd** [lsh0520/RGCL#6]
Thank you for your enlightening work. I am curious about the training of the rationale generator which does not seem to be detailedly mentioned in the paper. In section 4.1, it seems that the rationale generator is trained unsupervised, but popular IRD training methods need labels. Could you tell me how to train the rationale generator through unsupervised training?  Thank you.

**Comment by lsh0520** [comment]
Thank for your interest in our work.

Indeed, typical IRD trains in a supvervised fashion, while we borrowed its idea and tried to adapt it to the contrastive learning -- unsupervised learning.

Ovearlly, the gradient signal is generated by the the loss function as depicted in Equations(16-18) to perform instance-discrimination task (unsupervised learning). During this process, both the backbone encoder and the rationale generator is optimized to learn invariant features. 

And the key point here is how to enable the gradient back propagation to the rationale generator. Please see Equation (13) for details.

**Comment by Yzmshjd** [comment]
Thank you. I have got the main idea of the rationale generator's training.
